### PR TITLE
onionshare: init at 2.2

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Motivation for this change
Meet #83029

###### Things done
- Add derivation
- Add myself as maintainer
- Link derivation in pkgs/top-level

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4407,6 +4407,15 @@
...
     }];
   };
+  lourkeur = {
+    name = "Louis Bettens";
+    email = "louis@bettens.info";
+    github = "lourkeur";
+    githubId = 15657735;
+    keys = [{
+      longkeyid = "ed25519/0xDFE1D4A017337E2A";
+      fingerprint = "5B93 9CFA E8FC 4D8F E07A  3AEA DFE1 D4A0 1733 7E2A";
```
Comment (cole-h) [inline maintainers/maintainer-list.nix:4417]:
```suggestion
      fingerprint = "5B93 9CFA E8FC 4D8F E07A  3AEA DFE1 D4A0 1733 7E2A";
    }];
```

This (should) fix ofborg :)

File: pkgs/applications/networking/onionshare/default.nix [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      tor
+      obfs4
+    ];
+
+    patches = [ ./fix-paths.patch ];
+
+    # replace @tor@, @obfs4@, @geoip@ w/ store paths.
+    inherit tor obfs4;
+    inherit (tor) geoip;
+    postPatch = "substituteAllInPlace onionshare/common.py";
```
Comment (cole-h) [inline pkgs/applications/networking/onionshare/default.nix:77]: This should work as-is, but please test first... I've been known to make mistakes.

```suggestion
    patches = [
      (substituteAll {
        src = ./fix-paths.patch;
        inherit tor obfs4;
        inherit (tor) geoip;
      })
    ];
```

Comment (lourkeur) [reply pkgs/applications/networking/onionshare/default.nix:77]: That breaks my code: the patch references `@out@`.

Comment (lourkeur) [reply pkgs/applications/networking/onionshare/default.nix:77]: What is the goal of your requested change?

Comment (cole-h) [reply pkgs/applications/networking/onionshare/default.nix:77]: The goal is to make it look less cluttered and follow the soft standard of what happens in most other packages.

To deal with `@out@`, the best way would be to keep `postPatch`, but change it to `substituteInPlace onionshare/common.py --subst-var out`.

Comment (lourkeur) [reply pkgs/applications/networking/onionshare/default.nix:77]: gotcha

**Comment by emmanuelrosa** [comment]
It worked for me.

**Comment by filalex77** [comment]
Result of `nixpkgs-review pr 85765` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
<br>- onionshare
<br>- onionshare-gui
</details>

**Review by Br1ght0ne — APPROVED** [review]

**Review by cole-h — APPROVED** [review]
Result of `nixpkgs-review pr 85765` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
<br>- onionshare
<br>- onionshare-gui
</details>

---

I ran `onionshare-gui`, which connected me to Tor. Clicking the `Test connection to Tor` button did what it says on the tin.

**Pull request merged**